src/Helper/Installation.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace Castor\Helper;
+
+use Psr\Cache\CacheItemPoolInterface;
+use Symfony\Component\DependencyInjection\Attribute\Exclude;
+use Symfony\Component\Process\Process;
+use Symfony\Contracts\Cache\CacheInterface;
+
+/**
+ * @internal
+ */
+#[Exclude]
+final class Installation
+{
+    private ?Architecture $architecture = null;
+    private ?InstallationMethod $method = null;
+    private ?string $path = null;
+
+    public function __construct(
+        private readonly CacheItemPoolInterface&CacheInterface $cache,
+    ) {
```

Comment:
let's do the detect in constructor, if it cost too much we can always use a lazy object.